src/string_scanner.cr

```diff
@@ -73,7 +75,7 @@ class StringScanner
 
   # Returns the current position of the scan offset.
   def offset : Int32
-    @str.byte_index_to_char_index(@byte_offset).not_nil!
+    @byte_offset
```

Comment:
Isn't this representative of the character offset, and not `byte_offset`?

Reply:
I think so, but it raises error when calling `#read_byte` to a multibyte character then calling `#offset` in the current implementation.
I concern this behavior is expected or not.

```crystal
require "string_scanner"

s = StringScanner.new("あ")
s.read_byte
s.offset #=> Unhandled exception: Nil assertion failed (NilAssertionError)
```

Reply:
Right, I think this change would be a breaking change so in theory we can'd do this.

However, I consider the existing definition of `offset` to be incorrect. `offset` should actually return the byte offset because that's more useful, and it's the only correct thing we can return if one can advance byte per byte. So we can consider this change a bugfix instead of a breaking change.

Reply:
Useful to who? Isn't it pretty useful to use the same index values when parsing `String`s as the `String`s themselves use when indexing with `String#[]`?

Reply:
Also, the suggested change is inconsistent with `offset=`